### ISSUE TITLE
VMS: Crew Corner resource documents list.

### ISSUE THREAD

**Issue by WaBirder57** [washingtontrails/vms#550]
Currently we do not have a place in VMS to share documents with Crew Leaders and ACLs that are not specific to a work party.  An example would be a link to How To documents for CL/ACL's or Incident Report Forms, Crew Leader Handbook, etc.

Is it possible to add a new item in My Backpack that would link to shared google documents foler which would be visible for CL, ACL and Admin. This could be called "Resources" . We don't want to post these on our website where they would be accessible to the public.  Conceivably this would link to our shared Google Folder where we place these documents for anyone who has the link can access.  Here is an example: https://drive.google.com/a/wta.org/folderview?id=0B1o6JKvh8eh-YkNvQ0FON2dyZVE&usp=sharing#

We are open to other suggested ways to create this but we wanted something easy to do so we could have it for launch and still flexible so we can add and updated documents.

Here is the current My Backpack Crew Corner List of itmes where we would add this new link for resrouces.
![mybackpackcrewleader](https://cloud.githubusercontent.com/assets/8312206/12428469/87f6f452-be99-11e5-979e-94ee3cccc711.JPG)

**Comment by davisagli** [comment]
The solution I would recommend is:
1. Have us set things up so that users with a VMS Permission Level of Crew Leader or Assistant Crew Leader automatically get placed in a Plone "Leaders" group.
2. Set up a Leader Resources folder in Plone that is private, but has read access shared with the Leaders group.

How does that sound? A google drive folder would work, but requires you to set up access for each user separately, and is not searchable as part of your website.

**Comment by WaBirder57** [comment]
Sounds like your suggestion using Plone groups is the way to go.  In talking with Loren I learned this came up during our Plone training.

We would like to have Plone groups for the following 3 types of users as identified in Salesforce.
Members
ACL or CL
Hiking Guide Correspondent

As I understand this, membership in these plone groups would change based on the values of the corresponding Salesforce Contact fields.

SF Contact Fields are:
Hiking_Guide_Correspondent__c = "True"
Household_Membership_Status__c = "Current" or "Grace Period"
VMS_Permission_Level__c = "Crew Leader" or "Assistant Crew Leader"

**Comment by davisagli** [comment]
Okay, I've set up these automatic groups on mybackpack.wta.org and created a folder at https://mybackpack.wta.org/crew-leader-corner/resources which is shared to be readable by the CLs & ACLs group, and linked from the My Backpack sidebar.

Once you've confirmed that it works as expected, let me know and I'll set up a similar folder on www.wta.org.

Remember to leave the folder private, NOT publish it.

**Comment by WaBirder57** [comment]
I have added several documents to this folder and it looks like it works as you describe.  

I'm not that familiar with using Groups in Plone, but should I expect to see a list of Group Members when I am at this page: https://mybackpack.wta.org/@@usergroup-groupmembership?groupname=WPLeaders

What I see is this:
![acl and cl group](https://cloud.githubusercontent.com/assets/8312206/12456447/b38abf52-bf54-11e5-954a-d056338036cd.JPG)

**Comment by davisagli** [comment]
You would see a list of users there for normal, manually managed groups. These automatic groups don't provide a list of all users in the group because Plone doesn't provide an efficient way to query for all users with a particular member property (the list would take minutes to load). But when looking at one particular user it will consider it to be part of the group when it matches the criteria.

**Comment by WaBirder57** [comment]
I did a quick check and found that Loren@wta.org was a member of this ACL and CL group.  When I look at her salesforce record(s) (.VMS sandbox as well as production) she is set with a VMS permission level of Volunteer.  How would she have become a member of this plone group?

![acl and cl group loren wta org](https://cloud.githubusercontent.com/assets/8312206/12457396/145575da-bf59-11e5-9592-3811f0cf98fd.JPG)

**Comment by davisagli** [comment]
I think you misunderstood the search on that page. It is meant for manually managed groups and is how you find someone to add to the group. The fact that it found Loren when you searched does not mean she is in the group.

**Comment by WaBirder57** [comment]
So how do i confirm who is in which group?  These manual groups don't show up on the User Overview display.

**Comment by davisagli** [comment]
Ah...Plone's default profile form has a tab that links to a list of the user's groups, but that isn't included in your customized profile form. Shall I add it for Managers only?

**Comment by WaBirder57** [comment]
Ah indeed!  Yes Please and thank you.

**Comment by davisagli** [comment]
Okay, I added a list of groups on the right side of the profile form, for managers.

**Comment by WaBirder57** [comment]
Thanks, that helps.  

How did assign the Group for "CL and ACL's" to have access to the http://mybackpack.wta.org/crew-leader-corner/resources folder?

**Comment by davisagli** [comment]
I did it programmatically. We can show you how to do it for other folders using Plone's Sharing tab once we re-enable the Sharing tab, but I'd rather handle that as a support ticket after VMS launch if that's all right.

**Comment by WaBirder57** [comment]
OK got it, works for launch.  I assume it is Read Only access.

Post launch when we set up Plone folders for the other groups Members and Hiking Guide Correspondents what can I tell staff about how they will be able to control this on our production site?

**Comment by davisagli** [comment]
We should probably give you some training on the Sharing tab, eh @skleinfeldt?

**Comment by WaBirder57** [comment]
Testing this I'm not sure if it is working for andrea@wta.org.

Her SF Record: https://cs9.salesforce.com/003K0000016Tuxv?srPos=0&srKp=003 has VMS permissions set to Admin.

Her profile: http://mybackpack.wta.org/@@user-information?userid=andrea%40wta.org doesn't show her as a member of the ACL and CL group (perhaps that is correct) but she doesn't have access to the Resources Tab.  

Since she is not a Manager on the mybackpack.wta.org site.  Perhaps we should add to your group assignment code that if VMS Permission field =  Admin, Crew Leader or Assistant Crew Leader.

**Comment by davisagli** [comment]
You're right; I just adjusted it to put VMS Permission Level = Admin in the group too.
